src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class for parsing strings into commands.
+ */
+public class Parser {
+    class EmptyInputException extends RuntimeException {
+        public EmptyInputException() {
+            super("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+
+    class InvalidInputException extends RuntimeException {
+        public InvalidInputException() {
+            super("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Parses a string into a command.

```

Comment:
I like how your summary statements are all short and clear and starts with a verb. This makes the comments very easy to read!